metabase/metabase
# Milestone 1 of [Epic] Prevent auto-apply filters in slow dashboards

This closes milestone 1 of https://github.com/metabase/metabase/issues/29267

### Description

Provide a toggle on dashboards to set the Auto-apply filters behavior. When this toggle is off, when users apply any filter, instead of automatically refreshing the whole dashboard, an **Apply button** will appear and the dashboard will only refresh when you click this apply button.

### How to verify

Test
1. App dashboards
2. X-ray dashboards
3. Public dashboards
4. Signed dashboard embeddings
5. Full-app embeddings

That with auto-apply filters on or off, they still work as expected. That is
1. If `auto_apply_filters: true` (default), all dashboards should refresh every time a filter changes.
2. If `auto_apply_filters: false` is set, all dashboards should display apply button after a filter is changed.

### Demo
https://user-images.githubusercontent.com/1937582/233137338-343cfd5e-27ef-4c0a-bd7b-1db03f79c6b6.mov

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30232)
